modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -2653,6 +2653,8 @@ private CompletableFuture<Boolean> ensureReplicaIsPrimary(ReplicaRequest request
             expectedTerm = ((ReadOnlyDirectReplicaRequest) request).enlistmentConsistencyToken();
 
             assert expectedTerm != null;
+        } else if (request instanceof BuildIndexReplicaRequest) {
+            expectedTerm = ((BuildIndexReplicaRequest) request).enlistmentConsistencyToken();
```

Comment:
If got term doesn't match an expected one, you will get PrimaryReplicaMissException. How do you handle it?

Reply:
No way, the index building task will receive an error, log it and stop working.

Reply:
Well, such approach is incorrect. Index builder should gracefully handle such exceptions by retrying await -> send -> get flow.

Reply:
Discussed in person: in `IndexBuilder` the `PrimaryReplicaMissException` will be handled only by logging at the DEBUG level, a new attempt to start building the index will be handled by `PrimaryReplicaEvent#PRIMARY_REPLICA_ELECTED`.